palace/models/waveportoperator.cpp

```diff
@@ -1021,8 +1025,12 @@ std::complex<double> WavePortData::GetPower(mfem::ParComplexGridFunction &E,
   pi.AddBoundaryIntegrator(new VectorFEBoundaryLFIntegrator(nxHi_func), attr_marker);
   pr.UseFastAssembly(false);
   pi.UseFastAssembly(false);
+  pr.UseDevice(false);
+  pi.UseDevice(false);
   pr.Assemble();
   pi.Assemble();
+  pr.UseDevice(true);
+  pi.UseDevice(true);
```

Comment:
The flag set by `UseDevice` is just an execution flag, rather than a memory location flag. So this is just a minor optimization so that operations like `pr = 0.0` inside of `LinearForm::Assemble` happen on the host since we know assembly is going to happen on the host anyway. No need to move the `Vector` to the device only to pull it back after setting it equal to a value.